src/ops/ops.go

```diff
@@ -503,3 +506,14 @@ func (o *ops) GetMustGatherLogs(workDir string, kubeconfigPath string) (string,
 	}
 	return path.Join(workDir, tarName), nil
 }
+
+func (o *ops) CreateRandomHostname(hostname string) error {
+	command := fmt.Sprintf("echo %s > /etc/hostname", hostname)
```

Comment:
If it's supposed to run under sudo with a non-root user, then please make sure you test it. `echo` is known not to work with sudo. `sudo bash -c echo ... > file" didn't work for me either. A good way to echo to a file that requires sudo is `echo "text" | sudo tee --append /path/to/file`

Reply:
echo by itself indeed does not work , but running it via nsenter with bin/bas -c works just fine. Checked it manually. Also , it it was not working, then installation would have stuck